pysmt/solvers/pico.py

```diff
@@ -0,0 +1,133 @@
+#
+# This file is part of pySMT.
+#
+#   Copyright 2014 Andrea Micheli and Marco Gario
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+import picosat
+
+import pysmt.logics
+
+from pysmt import typing as types
+from pysmt.solvers.solver import Solver
+from pysmt.solvers.eager import EagerModel
+from pysmt.cnf import CNFizer
+from pysmt.decorators import clear_pending_pop
+
+
+class PicosatSolver(Solver):
+    """PicoSAT solver"""
+
+    LOGICS = [ pysmt.logics.QF_BOOL ]
+
+    def __init__(self, environment, logic, options=None):
+        Solver.__init__(self,
+                        environment=environment,
+                        logic=logic,
+                        options=options)
+
+        self.mgr = environment.formula_manager
+        self.pico = picosat.picosat_init()
+        self.converter = None
+        self.cnfizer = CNFizer(environment=environment)
+        self.latest_model = None
+        self._var_ids = {}
+
+
+    def _get_var_id(self, symbol):
```

Comment:
Not sure. The whole converter design is not applicable to picosat as there is no concept of foumula. All you can do is assert clauses (and the index of the variables must be specified taking care of ```picosat.picosat_inc_max_var(self.pico)```) hence I think it is much more natural to have it here...